commit b0fc5e043401df4cd243352f1030c4d23e767347
Author: Takashi Iwai <tiwai@suse.de>
Date:   Fri Nov 21 08:37:03 2008 +0100

    ALSA: hda - Add a quirk for Dell Studio 15
    
    Added the matching model=dell-m6 for Dell Studio 15 laptop.
    
    Signed-off-by: Takashi Iwai <tiwai@suse.de>

Adjusted to apply to Debian's 2.6.26 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.26.orig/sound/pci/hda/patch_sigmatel.c linux-source-2.6.26/sound/pci/hda/patch_sigmatel.c
--- linux-source-2.6.26.orig/sound/pci/hda/patch_sigmatel.c	2009-01-29 16:43:51.000000000 -0700
+++ linux-source-2.6.26/sound/pci/hda/patch_sigmatel.c	2009-02-08 23:36:08.000000000 -0700
@@ -1308,6 +1308,8 @@ static struct snd_pci_quirk stac92hd73xx
 				"unknown Dell", STAC_DELL_M6),
 	SND_PCI_QUIRK(PCI_VENDOR_ID_DELL, 0x0271,
 				"unknown Dell", STAC_DELL_M6),
+	SND_PCI_QUIRK(PCI_VENDOR_ID_DELL, 0x029f,
+				"Dell Studio 15", STAC_DELL_M6),
 	{} /* terminator */
 };
 
